Makefile

```diff
@@ -0,0 +1,75 @@
+#Python3 and virtual environment
+PY3 := $(shell command -v python3 2> /dev/null)
+VENV := .venv
+PYTHON3 := $(VENV)/bin/python3
+
+#If caramel_tool exists in the venv caramel has been installed
+INSTALLED := $(VENV)/bin/caramel_tool
```

Comment:
Better to give this a name "CARAMEL_TOOL"  instead. That makes reading the rules below out-of-context easier.